tests/utilities/test_data.py

```diff
@@ -55,6 +58,20 @@ def _check_warning_raised(data, expected):
     batch = [{"test": torch.zeros(10, 10), "test_1": torch.zeros(11, 10)}]
     _check_warning_raised(batch, 10)
 
+    # Error raised
+    batch = "test string"
+    _check_error_raised(batch)
+
+    data = {"test": ["some text"] * 7}
+    _check_error_raised(data)
+
+    class CustomBatchCollector:
```

Comment:
What is this CustomBatchCollector has a len ?

Reply:
It's just a custom batch class.I'll update the name

Reply:
I meant, a `CustomBatch` object could have a `__len__` defined as the `batch_size`. Should we consider this too ?

Reply:
not sure if we should do this. I haven't seen any case where user have a custom batch structure and have defined `__len__`. Plus we have `__len__` inside PT dataset so it might confuse users more.

Reply:
I have seen several actually.